pkg/controller/spire-oidc-discovery-provider/configmaps.go

```diff
@@ -106,3 +105,15 @@ server {
 
 	return configMap, nil
 }
+
+// stripProtocol removes "http://" or "https://"" from the beginning of a string.
+// If no protocol prefix is found, it returns the original string unmodified.
+func stripProtocol(url string) string {
+	if strings.HasPrefix(url, "https://") {
```

Comment:
A validation check is required for case-sensitivity of the jwt issuer fields during jwt token creation and oidc discovery document fields.